src/zenml/models/v2/core/curated_visualization.py

```diff
@@ -0,0 +1,254 @@
+#  Copyright (c) ZenML GmbH 2025. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Models representing curated visualizations."""
+
+from typing import TYPE_CHECKING, Optional
+from uuid import UUID
+
+from pydantic import Field
+
+from zenml.enums import CuratedVisualizationSize, VisualizationResourceTypes
+from zenml.models.v2.base.base import BaseUpdate
+from zenml.models.v2.base.scoped import (
+    ProjectScopedRequest,
+    ProjectScopedResponse,
+    ProjectScopedResponseBody,
+    ProjectScopedResponseMetadata,
+    ProjectScopedResponseResources,
+)
+
+if TYPE_CHECKING:
+    from zenml.models.v2.core.artifact_version import ArtifactVersionResponse
+
+
+# ------------------ Request Model ------------------
+
+
+class CuratedVisualizationRequest(ProjectScopedRequest):
+    """Request model for curated visualizations.
+
+    Each curated visualization is linked to exactly one resource of the following
+    types:
+    - **Deployments**: Surface visualizations on deployment dashboards
+    - **Models**: Highlight evaluation dashboards and monitoring views next to
+      registered models
+    - **Pipelines**: Associate visualizations with pipeline definitions
+    - **Pipeline Runs**: Attach visualizations to specific execution runs
+    - **Pipeline Snapshots**: Link visualizations to snapshot configurations
+    - **Projects**: Provide high-level project dashboards and KPI overviews
+
+    To attach a visualization to multiple resources, create separate curated
+    visualization entries for each resource.
+    """
+
+    artifact_version_id: UUID = Field(
+        title="The artifact version ID.",
+        description="Identifier of the artifact version providing the visualization.",
+    )
+    visualization_index: int = Field(
+        ge=0,
+        title="The visualization index.",
+        description="Index of the visualization within the artifact version payload.",
+    )
```

Comment:
Is there any reason why we're using the artifact version ID and visualization index to reference the visualization, instead of the ID of the visualization itself (`ArtifactVisualizationSchema`)?